### DISCUSSION TITLE
How can I add my customize page in the menu?

### DISCUSSION THREAD

**Discussion by dingsheng999888** [apache/superset#10618] [General]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
I have try to add a view in the menu behind Dashboard,but superset can not add my view with permission.

**Describe alternatives you've considered**

**Additional context**
        appbuilder.add_view(
            MyView,
            "MyView",
            label=__("MyView"),
            icon="fa-dashboard",
            category="MyView",
            category_icon="",
        )

**Comment by memorylorry** [comment]
Hi, bro. Have you solved this problem. I meet it, too.

**Comment by memorylorry** [comment]
I have find the way to solve this problem. After we added this view into `appbuilder`, metadata need to be added into `ab_permission` and `ab_permission_view` tables by myself. The left requirements is to make sense what it is permission and what it is permission_view.

**Reply by skyicrus** [reply]
What files have you modified, what changes have been made to the front end and the back end

**Comment by mistercrunch** [comment]
I think it may be possible to use the hook here https://github.com/apache/superset/blob/master/superset/config.py#L1177-L1181, and access `app.appbuilder` to call things like` app.appbuilder.add_link` https://github.com/apache/superset/blob/master/superset/initialization/__init__.py#L231-L236

Though I think of the calls matters, so you would only be able to append to existing menus or create new categories.

Could be good for someone to test this and document it.
